pkg/async/pool/pool.go

```diff
@@ -273,3 +255,27 @@ type unit struct {
 	Context context.Context
 	Runner  async.Runner
 }
+
+type loggingScheduler struct {
+	Inner Scheduler
+	Name  string
+}
+
+func (w *loggingScheduler) Schedule(ctx context.Context, r async.Runner) error {
+	return w.log(ctx, w.Inner.Schedule(ctx, async.Func(func(ctx context.Context) error {
+		return w.log(ctx, r.Run(ctx))
+	})))
+}
+
+// WithLogging creates a scheduler which logs the errors returned from the scheduling as well as executing phase
```

Comment:
Does this mean that if a service update gobox without switching over to `WithLogging` they will lose logging without warning?

If that's the case, I think that we should make this `WithoutLogging` and make the default include logging.

Reply:
Yeah, looks like this is in use only in mailroomapi and ppsgateway.  So, if you are willing to manually upgrade those repos, then this is ok.

Reply:
Sounds good to me. Thanks for looking into it!